tests/test_push.py

```diff
@@ -1262,6 +1263,107 @@ def test_classify_almost_bad_push(
+                        ]
+                    }, 
+                intermittent={}, 
+                unknown={}
+                ),
+            False, # does not perform retrigger
+            ),
+
+
+        (
+            Regressions(real={}, 
+                intermittent={}, 
+                unknown={}),
+            False, # does not perform retrigger
+        ),
+
+        (
+            Regressions(
+                real={},
+                intermittent={},
+                unknown={
+                    "group1": [
+                        Task.create(
+                            id="NjJqN07WQ9Cs6HvVLUJXnw",
+                            label=f"test-task{id}",
+                            result="failed",
+                            _results=[GroupResult(group=f"group{id}", ok=False)],
+                        )
+                    ]
+                },
+            ),
+            True, # performs retrigger
+        ),
+
+        (
+            Regressions(
+                real={},
+                intermittent={
+                    "group1": [
+                        Task.create(
+                            id="NjJqN07WQ9Cs6HvVLUJXnw",
+                            label=f"test-task{id}",
+                            result="failed",
+                            _results=[GroupResult(group=f"group{id}", ok=False)],
+                        )
+                    ]
+                },
+                unknown={},
+            ),
+            False, # does not perform retrigger
+        ),
+    ],
+)
+def test_classify_should_retrigger(
+    responses, monkeypatch, classify_regressions_return_value, expected_result
+):
+    rev = "a" * 40
+    branch = "autoland"
+    push = Push(rev, branch)
+
+    def mock_return(self, *args, **kwargs):
+        return classify_regressions_return_value
+
+    monkeypatch.setattr(Push, "classify_regressions", mock_return)
+    responses.add(
+        responses.PUT,
+        "https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/JHT2zBEmRvKeXTAs0_PXYQ",
+        json={"payload": {}, "tags": {"retrigger": "true", "label": "test_retrigger"}},
+        status=200,
+    )
+
+    responses.add(
+        responses.GET,
+        "https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/NjJqN07WQ9Cs6HvVLUJXnw",
+        json={"payload": {}, "tags": {"retrigger": "true", "label": "test_retrigger"}},
+        status=200,
+    )
+
+    create_new_task_url_matcher = re.compile(
+        r"https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/*"
+    )
+
+    responses.add(responses.PUT, create_new_task_url_matcher, status=200)
+
+    push.classify(should_retrigger=True)
+
+    # verify last call was to create a new task
+    assert (len(responses.calls) != 0 and responses.calls[-1].request.method == "PUT") == expected_result
+    assert (len(responses.calls) != 0 and bool(create_new_task_url_matcher.match(responses.calls[-1].request.url))) == expected_result
```

Comment:
Instead of just checking that we retrigger, it might be worth checking that we retrigger the right things